core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala

```diff
@@ -324,4 +329,29 @@ class ZkMigrationIntegrationTest {
       assertTrue(firstProducerIdBlock.firstProducerId() < producerIdBlock.firstProducerId())
     }
   }
+
+  /**
+   * Start a KRaft cluster with migrations enabled, verify that the controller does not accept metadata changes
+   * through the RPCs
+   */
```

Comment:
We should also explain in this comment why the migration does not proceed beyond premigration (I assume it's because we're waiting for brokers and we don't have any due to brokers =  0)